src/primitives.jl

```diff
@@ -210,8 +210,18 @@ dropnull!(X::NullableVector) = deleteat!(X, find(X.isnull)).values # -> Vector
 
 Returns whether or not any entries of `X` are null.
 """
-anynull(X::Any) = any(_isnull, X)           # -> Bool
-anynull(X::NullableArray) = any(X.isnull)   # -> Bool
+anynull(X::Any) = any(_isnull, X)                              # -> Bool
+anynull(X::NullableArray) = any(X.isnull)                      # -> Bool
+function anynull{T}(X::AbstractArray{T})                       # -> Bool
+    u = isa(T, Union)
+    if u && !any(f -> eval(:($T.$f)) <: Nullable || Nullable <: eval(:($T.$f)), fieldnames(T))
```

Comment:
We all missed this case when implementing the short-circuit for dropnull & dropnull!, but Unions are tricky to handle. Any ideas how to do this without the eval? I couldn't figure out how to iterate over the types of the union without it.